### DISCUSSION TITLE
Error: Failed to create chaser

### DISCUSSION THREAD

**Discussion by Rhinoceroscity** [Autodesk/maya-usd#1674] [Q&A]
Hey everyone, I'm trying to get a custom export chaser to work in its own plugin. I've gotten it to work by modifying the Maya-USD plugin directly and adding the chaser there, but if its part of its own plugin, I simply get an error that it cannot create the chaser. The plugin compiles and loads just fine. I've attached the main.cpp file for my little chaser example plugin (which again, if its part of the Maya-USD plugin directly, works fine), and I make sure the load order is the Pixar USD plugin, the Maya USD plugin, then my own chaser plugin. I'm using Maya 2022.

```
#define HAVE_ROUND

#pragma warning(disable:4244 4305)
#include <mayaUsd/fileio/chaser/exportChaserRegistry.h>
#include <mayaUsd/utils/util.h>

#include <pxr/base/tf/stringUtils.h>
#include <pxr/base/vt/dictionary.h>
#include <pxr/pxr.h>
#include <pxr/usd/usd/primRange.h>
#pragma warning(default:4244 4305)

#include <maya/MFnDependencyNode.h>
#include <maya/MFnStringData.h>
#include <maya/MFnTypedAttribute.h>
#include <maya/MGlobal.h>
#include <maya/MStatus.h>
#include <maya/MFnPlugin.h>

PXR_NAMESPACE_OPEN_SCOPE

class MyExportChaser : public UsdMayaExportChaser
{
public:
    MyExportChaser(
        UsdStagePtr stage,
        const UsdMayaExportChaserRegistry::FactoryContext::DagToUsdMap& dagToUsd)
        : m_stage(stage)
        , m_dagToUsd(dagToUsd)
    {

    }

    bool ExportDefault() override
    {
        for (const auto it : m_dagToUsd)
        {
            MGlobal::displayInfo(it.first.fullPathName());
        }
        return true;
    }

private:
    UsdStagePtr m_stage;
    const UsdMayaExportChaserRegistry::FactoryContext::DagToUsdMap& m_dagToUsd;
};

PXRUSDMAYA_DEFINE_EXPORT_CHASER_FACTORY(mychaser, ctx)
{
    std::map<std::string, std::string> myArgs;
    TfMapLookup(ctx.GetJobArgs().allChaserArgs, "mychaser", &myArgs);

    return new MyExportChaser(ctx.GetStage(), ctx.GetDagToUsdMap());
}

PXR_NAMESPACE_CLOSE_SCOPE

__declspec(dllexport) MStatus initializePlugin(MObject obj)
{
    MFnPlugin plugin(obj);

    return MStatus::kSuccess;
}

__declspec(dllexport) MStatus uninitializePlugin(MObject obj)
{
    return MStatus::kSuccess;
}
```

Here's the command I use (MayaUSDTest is my own plugin containing the chaser):

```
cmds.loadPlugin('pxrUsd')
cmds.loadPlugin('mayaUsdPlugin')
cmds.loadPlugin('MayaUSDTest')

cmds.mayaUSDExport(
    file=r"C:\users\myuser\desktop\test.usda",
    chaser=['mychaser'],
    chaserArgs=[])
```

And finally the error, in case it helps at all.

```
# Opening layer 'C:/users/myuser/desktop/test.usda' for writing # 
# Saving stage # 
# Error: Failed to create chaser: mychaser # 
```

**Comment by kxl-adsk** [comment]
Sorry for the very late response. You might have solved it already, but two things come to my mind:
1. You have to make sure that `mayaUsdPlugin` is loaded first before PXRUSDMAYA_DEFINE_EXPORT_CHASER_FACTORY executes
2. We have recently added python bindings to the translation framework. You can write your chaser in python if that's simpler and acceptable from a performance perspective. See example - https://github.com/Autodesk/maya-usd/blob/83887106d8ff21c68607fdf528b97e4506491a99/test/lib/mayaUsd/fileio/testExportChaser.py
